### ISSUE TITLE
"display all" results in Warning

### ISSUE THREAD

**Issue by matthiasradde** [phpmyadmin/phpmyadmin#13031]
### Steps to reproduce
Occurs on tables and views, but will write only "table" as it is easier/shorter
Occurs with metro-theme and german as well as with the default-theme and english

1. created a table with more than 25 rows (I used on with 2 columns and 32 rows), table-definition as an export within phpMyAdmin (of course the column names are in backticks but displayed as monospace-font here):
```
CREATE TABLE `kb_tags` (
  `id` int(10) UNSIGNED NOT NULL,
  `tag` varchar(255) NOT NULL
) ENGINE=MyISAM DEFAULT CHARSET=latin1;
ALTER TABLE `kb_tags`
  ADD PRIMARY KEY (`id`),
  ADD KEY `tag` (`tag`);
ALTER TABLE `kb_tags`
  MODIFY `id` int(10) UNSIGNED NOT NULL AUTO_INCREMENT;
```
2. display the content of the table - the first 25 rows are displayed
3. select the "display all"-checkbox

### Expected behaviour
Tell us what should happen
display all 32 rows

### Actual behaviour
Tell us what happens instead
display all 32 rows plus the following warning below the result-table:
 - all entries are display plus a warning at the bottom of the screen (the result-pane):

```
Warning in .\libraries\DisplayResults.php#869
A non-numeric value encountered

Backtrace

.\libraries\DisplayResults.php#4933: PMA\libraries\DisplayResults->_getTableNavigation(
integer 0,
integer 0,
boolean false,
string '<form action="sql.php" method="post" class="print_ignore"> <input type="hidden" name="db" value="lsy_kb4" /><input type="hidden" name="table" value="kb_tags" /><input type="hidden" name="token" value="b2685adb640ed0549522ea800576f7cf" /><input type="hidden" name="sort_by_key" value="1" />Sort by key: <select name="sql_query" class="autosubmit"> <option value="SELECT * FROM `kb_tags` ORDER BY `id` ASC">PRIMARY (ASC)</option><option value="SELECT * FROM `kb_tags` ORDER BY `id` DESC">PRIMARY (DESC)</option><option value="SELECT * FROM `kb_tags` ORDER BY `tag` ASC">tag (ASC)</option><option value="SELECT * FROM `kb_tags` ORDER BY `tag` DESC">tag (DESC)</option><option value="SELECT * FROM `kb_tags` " selected="selected">None</option></select> </form> ',
)
.\libraries\DisplayResults.php#4378: PMA\libraries\DisplayResults->_getPlacedTableNavigations(
integer 0,
integer 0,
string 'top_direction_dropdown',
boolean false,
string '<form action="sql.php" method="post" class="print_ignore"> <input type="hidden" name="db" value="lsy_kb4" /><input type="hidden" name="table" value="kb_tags" /><input type="hidden" name="token" value="b2685adb640ed0549522ea800576f7cf" /><input type="hidden" name="sort_by_key" value="1" />Sort by key: <select name="sql_query" class="autosubmit"> <option value="SELECT * FROM `kb_tags` ORDER BY `id` ASC">PRIMARY (ASC)</option><option value="SELECT * FROM `kb_tags` ORDER BY `id` DESC">PRIMARY (DESC)</option><option value="SELECT * FROM `kb_tags` ORDER BY `tag` ASC">tag (ASC)</option><option value="SELECT * FROM `kb_tags` ORDER BY `tag` DESC">tag (DESC)</option><option value="SELECT * FROM `kb_tags` " selected="selected">None</option></select> </form> ',
)
.\libraries\sql.lib.php#1685: PMA\libraries\DisplayResults->getTable(
,
array,
array,
boolean false,
)
.\libraries\sql.lib.php#1976: PMA_getHtmlForSqlQueryResultsTable(
,
string './themes/original/img/',
NULL,
array,
boolean true,
integer 32,
integer 32,
NULL,
,
array,
)
.\libraries\sql.lib.php#2199: PMA_getQueryResponseForResultsReturned(
,
array,
string 'lsy_kb4',
string 'kb_tags',
NULL,
NULL,
,
string './themes/original/img/',
integer 32,
integer 32,
NULL,
NULL,
NULL,
NULL,
NULL,
string 'SELECT * FROM `kb_tags`',
NULL,
)
.\libraries\sql.lib.php#2061: PMA_executeQueryAndGetQueryResponse(
array,
boolean false,
string 'lsy_kb4',
string 'kb_tags',
NULL,
NULL,
NULL,
NULL,
NULL,
NULL,
string 'sql.php?db=lsy_kb4&amp;table=kb_tags&amp;sql_query=SELECT+%2A+FROM+%60kb_tags%60&amp;token=b2685adb640ed0549522ea800576f7cf',
string './themes/original/img/',
NULL,
NULL,
NULL,
string 'SELECT * FROM `kb_tags`',
NULL,
NULL,
)
.\sql.php#221: PMA_executeQueryAndSendQueryResponse(
array,
boolean false,
string 'lsy_kb4',
string 'kb_tags',
NULL,
NULL,
NULL,
NULL,
NULL,
NULL,
string 'sql.php?db=lsy_kb4&amp;table=kb_tags&amp;sql_query=SELECT+%2A+FROM+%60kb_tags%60&amp;token=b2685adb640ed0549522ea800576f7cf',
string './themes/original/img/',
NULL,
NULL,
NULL,
string 'SELECT * FROM `kb_tags`',
NULL,
NULL,
)
```

### Server configuration
**Operating system**:
Windows 7 x64

**Web server:**
Apache 2.4.25 x64

**Database:**
MySQL 5.6.35 x64

**PHP version:**
7.1.2 x64

**phpMyAdmin version:**
4.6.6

### Client configuration
**Browser:**
Firefox 51.0.1 x32

**Operating system:**
Windows 7 x64

**Comment by nijel** [comment]
Param | Value 
 -----------|--------------------
 Error Type | Warning
 Error Message |A non-numeric value encountered
 Exception Type |php
 phpMyAdmin version |4.6.5.2
 Incident count | 1193
 Link | [Report#29892](https://reports.phpmyadmin.net/reports/view/29892)

*This comment is posted automatically by phpMyAdmin's [error-reporting-server](https://reports.phpmyadmin.net).*

**Comment by ben-sab** [comment]
This doesn't seem to be fixed in PHP 7.2.10 PhpMyAdmin 4.6.6 on Ubuntu 18, when clicking on "show all" in tables I get a very similar warning: 
`Warning in ./libraries/DisplayResults.php#869 A non-numeric value encountered`
`./libraries/DisplayResults.php#4934: PMA\libraries\DisplayResults->_getTableNavigation(`
...
`./libraries/DisplayResults.php#4379: PMA\libraries\DisplayResults->_getPlacedTableNavigations(`
...
`./libraries/sql.lib.php#1689: PMA\libraries\DisplayResults->getTable(`
...
`./libraries/sql.lib.php#1980: PMA_getHtmlForSqlQueryResultsTable(`
...
`./libraries/sql.lib.php#2199: PMA_getQueryResponseForResultsReturned(`
...
`./libraries/sql.lib.php#2079: PMA_executeQueryAndGetQueryResponse(`
...
`./sql.php#221: PMA_executeQueryAndSendQueryResponse(`

**Comment by williamdes** [comment]
@ben-sab Did you see :
> nijel added this to the 4.7.0 milestone on 1 Mar 2017

**Comment by ben-sab** [comment]
@williamdes thanks for the attention. Yes I did, I tried changing DisplayResults.php as mentioned but still the same outcome.

**Comment by linuxmangr** [comment]
Hello to all,
I user the Debian 9.6 stretch
PHP 7.2.13-1+0~20181207100540.13+stretch~1.gbpf57305
MariaDB 10.2.19
Phpmyadmin  4.6.6deb4

I found line when have problem "Display all"
I use the geany code editor on (Linux Mint) and change lines start from 
867 -> 881 with Comment out lines
```
       // Move to the next page or to the last one
        if ($this->__get('unlim_num_rows') === false // view with unknown number of rows
            || ($_SESSION['tmpval']['max_rows'] != self::ALL_ROWS
            && $_SESSION['tmpval']['pos'] + $_SESSION['tmpval']['max_rows'] < $this->__get('unlim_num_rows')
            && $this->__get('num_rows') >= $_SESSION['tmpval']['max_rows'])
        ) {

            $table_navigation_html
                .= $this->_getMoveForwardButtonsForTableNavigation(
                    $html_sql_query, $pos_next, $is_innodb
                );

        } // end move toward

        // show separator if pagination happen
```
After change restart apache2 and mysql (mariadb) 
Now I not see this Warning.

**Comment by linuxmangr** [comment]
> @ben-sab @linuxmangr Can you try with the latest version of phpMyAdmin ?

On this server when I found it after install all needed software I can't try. (Is Production)
On Test Server when I found other problematic Issues with files I try the new version.
I think the version on repositories  https://mirror.yandex.ru/debian/pool/main/p/phpmyadmin/ is old.
If I found some time I will try latest version and inform about Issues or not.
